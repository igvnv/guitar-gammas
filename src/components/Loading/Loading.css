.loading {
  text-align: center;
}

.loading__message {
  animation: loading__message_animation 0.8s infinite alternate-reverse;
  color: #3b3b3b;
  font-size: 14px;
}

@keyframes loading__message_animation {
  from {
    opacity: 1;
  }
  to {
    opacity: 0.6;
  }
}
