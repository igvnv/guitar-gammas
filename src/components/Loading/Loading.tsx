import React from 'react';

import './Loading.css';

type LoadingProps = {
  message?: String;
};

/**
 * Displays container with information about loading process.
 */
const Loading: React.FC<LoadingProps> = ({ message = 'Загрузка...' }) => {
  return (
    <div className="loading">
      <span className="loading__message">{message}</span>
    </div>
  );
};

export default Loading;
