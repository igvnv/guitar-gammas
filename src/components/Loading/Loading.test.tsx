import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Loading from './Loading';

describe('Loading component', () => {
  it('has default message', () => {
    const { getByText } = render(<Loading />);

    getByText('Загрузка...');
  });

  test('snapshot', () => {
    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot();
  });

  it('displays custom message', () => {
    const loadingMessage = 'Пожалуйста, подождите';
    const { getByText } = render(<Loading message={loadingMessage} />);

    getByText(loadingMessage);
  });
});
