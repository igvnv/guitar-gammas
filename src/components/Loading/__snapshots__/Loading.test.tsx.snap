// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component snapshot 1`] = `
<div>
  <div
    class="loading"
  >
    <span
      class="loading__message"
    >
      Загрузка...
    </span>
  </div>
</div>
`;
